files/synchronize.py

```diff
@@ -173,6 +173,8 @@
    - Expect that dest=~/x will be ~<remote_user>/x even if using sudo.
    - To exclude files and directories from being synchronized, you may add 
      C(.rsync-filter) files to the source directory.
+   - Does not use SSH agent, authentication must be done using keys using a
```

Comment:
For the record, I've been using Ansible's synchronize module to rsync some files keys from server A to server B, relying on my ssh-agent for authentication.  It worked fine.

(Often what breaks ssh agent auth is using 'sudo' to become root -- sudo clears SSH_AUTH_SOCK from the environment, with the default sudo configuration on Ubuntu.)